src/hazmat/lucas.rs

```diff
@@ -342,10 +342,11 @@ pub fn lucas_test<T: Integer>(
 
     // Find `d` and `s`, such that `d` is odd and `d * 2^s = n - (D/n)`.
     // Since `(D/n) == -1` by construction, we're looking for `d * 2^s = n + 1`.
-    let (s, d) = decompose(candidate);
+    let (s, d) = decompose(&candidate);
 
     // Some constants in Montgomery form
-
+    // TODO(dp): Cloning `candidate` should not be necessary. It's only needed because the
```

Comment:
I wonder if we should just remove `to_integer()`

Reply:
I tried that and it's not that easy actually, my comment is misleading: the call here to `new_params_vartime` takes
`candidate` by value so all later calls to `candidate.as_ref()` become "borrow of moved value". 
I also tried to replace `candidate.as_ref()` in the other calls with just `&T::one()` –– after all we
just need a `T` and the value doesn't matter –– but that fails for `BoxedUint`.

Maybe what we need here is a variant of `from_limb_like` that just takes the bit precision from a
`T` (and not the whole `T`) so that it works for `BoxedUint` as well.

Reply:
Maybe in some places, e.g. on line 338 above, it's possible to avoid casting `u64` to `T` and instead read the first limb from `candidate` and compare that. Not sure it works in general though.

Reply:
I'd leave the TODO and I'll try to come up with something here in the coming days.